controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -670,3 +679,19 @@ func generateUUID() string {
 	newUUID := uuid.New().String()
 	return newUUID[:8]
 }
+
+// fetchCephFilesystemSubVolumeGroupByOwnerReference fetches the CephFilesystemSubVolumeGroup by owner reference.
+func fetchCephFilesystemSubVolumeGroupByOwnerReference(ctx context.Context, k8sClient client.Client, owner metav1.ObjectMeta, cephFilesystemSubVolumeGroupList *rookCephv1.CephFilesystemSubVolumeGroupList, listOptions ...client.ListOption) (*rookCephv1.CephFilesystemSubVolumeGroup, error) {
+
+	err := k8sClient.List(ctx, cephFilesystemSubVolumeGroupList, listOptions...)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cephFilesystemSubVolumeGroup := range cephFilesystemSubVolumeGroupList.Items {
+		if cephFilesystemSubVolumeGroup.OwnerReferences[0].Name == owner.Name {
+			return &cephFilesystemSubVolumeGroup, nil
+		}
+	}
+	return nil, nil
+}
```

Comment:
1. Let's have this as a func on `r` so there will be no need to pass the client or the context
2. Please use the term `list` instead of `fetch` to align with k8s API terminology 
3. Don't assume a single-owner reference
4. Compare by UUID and not by name as UUID is never reused
5. Don't  return the result, fill in the provided list object